traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.Steering.tpl.html

```diff
@@ -296,47 +308,46 @@
                         <small class="input-diff" ng-show="settings.isRequest && open() && deliveryService.trResponseHeaders != dsCurrent.trResponseHeaders">Current Value: [ {{dsCurrent.trResponseHeaders}} ]</small>
                         <span ng-show="hasError(deliveryServiceForm.trResponseHeaders)" class="form-control-feedback"><i class="fa fa-times"></i></span>
                     </div>
                 </div>
 
                 <div class="form-group" ng-class="{'has-error': hasError(deliveryServiceForm.ccrDnsTtl), 'has-feedback': hasError(deliveryServiceForm.ccrDnsTtl)}">
-                    <label class="control-label col-md-2 col-sm-2 col-xs-12">
+                    <label class="control-label col-md-2 col-sm-2 col-xs-12" for="ccrDnsTtl">
                         <span uib-popover-html="label('ccrDnsTtl', 'desc')" popover-trigger="click" popover-placement="top" popover-append-to-body="true" popover-class="popover-class">{{label('ccrDnsTtl', 'title')}}</span>
                     </label>
                     <div class="col-md-10 col-sm-10 col-xs-12">
-                        <input name="ccrDnsTtl" type="number" class="form-control" ng-model="deliveryService.ccrDnsTtl" ng-pattern="/^\d+$/" autofocus>
-                        <small class="input-error" ng-show="hasPropertyError(deliveryServiceForm.ccrDnsTtl, 'pattern')">Whole Number</small>
+                        <input id="ccrDnsTtl" name="ccrDnsTtl" type="number" class="form-control" ng-model="deliveryService.ccrDnsTtl" min="0" step="1">
                         <small class="input-diff" ng-show="settings.isRequest && open() && deliveryService.ccrDnsTtl != dsCurrent.ccrDnsTtl">Current Value: [ {{dsCurrent.ccrDnsTtl}} ]</small>
                         <span ng-show="hasError(deliveryServiceForm.ccrDnsTtl)" class="form-control-feedback"><i class="fa fa-times"></i></span>
                     </div>
                 </div>
 
                 <div class="form-group" ng-class="{'has-error': hasError(deliveryServiceForm.infoUrl), 'has-feedback': hasError(deliveryServiceForm.infoUrl)}">
-                    <label class="control-label col-md-2 col-sm-2 col-xs-12">
+                    <label class="control-label col-md-2 col-sm-2 col-xs-12" for="infoUrl">
                         <span uib-popover-html="label('infoUrl', 'desc')" popover-trigger="click" popover-placement="top" popover-append-to-body="true" popover-class="popover-class">{{label('infoUrl', 'title')}}</span>
                     </label>
                     <div class="col-md-10 col-sm-10 col-xs-12">
-                        <input name="infoUrl" type="text" class="form-control" ng-model="deliveryService.infoUrl" ng-maxlength="255" autofocus>
+                        <input id="infoUrl" name="infoUrl" type="url" title="Must be a valid URL" class="form-control" ng-model="deliveryService.infoUrl" maxlength="255">
```

Comment:
i could be wrong but i think people use this for non url info :)

Reply:
actually, i think you should just switch type back to text

Reply:
The message in the `title` is meant to be sufficient. I could add a red text error underneath, though. Probably should, actually, for consistency's sake.

Are there really people who use the "Info URL" field for things that aren't URLs? With _three unlimited length description fields on the object_? That sounds like a horribly unintuitive misuse of the field.

Reply:
yeah if i look at our own production data, there are many values that are not a URL. so if you keep type=url, then it will disable the update button for a number of delivery services and it's hard to understand why the update button is disabled because of this - https://github.com/apache/trafficcontrol/issues/3202

Reply:
and yes, there are 3 long descrip fields (don't get me started) but at comcast, those are intended for certain data (again, don't get me started).

Reply:
The three values I've seen in Comcast's production environment that are not either an empty string or a valid URL are "TBD", "NA" and a single JIRA ticket number. "TBD" and "NA" should be obliterated - they aren't valid URLs and give us no information. The JIRA ticket number is one of hundreds of linked JIRA tickets, but all of the others are actual URLs instead of just the number. Even if this field wasn't explicitly for URLs, that should be changed for consistency and was likely an oversight/mistake in the first place.

I've also added a message like the other fields have that indicates when the entered value is not a valid URL (although I can't help but interject that validation of constant constraints is best left to the user-agent). The angular attributes of the "advanced" section can be tweaked such that it's open by default when any of the fields have errors. We could even go so far as to disable the "Hide Advanced" button in that case. I suppose I could roll that into this PR, or just make a note on that issue and leave it for another PR. Doesn't matter much to me which one, provided that's an acceptable solution.

Reply:
expanding the advanced section when there is a form validation error sounds like a good idea but i'm not sure about disabling the hide button (i hate making those forms too complicated) but, anyhow, let's save it for another PR.